### PR TITLE
fix nil pointer in addons test

### PR DESCRIPTION
**Pull request by mikedanese**
Remove unnecessary goroutine by using ssh.Start and ssh.Wait instead of ssh.Run (which just call Start then Wait). We can do all our stuff in the middle

@zmerlynn @bprashanth

### CODE REVIEW

**Comment by mikedanese** [comment]
Actually, i think this fixes the nil pointer but then it could just block forever....

**Comment by mikedanese** [comment]
Okay fixed

File: test/e2e/addon_update.go [hunk 1]
```diff
@@ -399,16 +399,17 @@ func writeRemoteFile(sshClient *ssh.Client, data, dir, fileName string, mode os.
...
-		fmt.Fprintf(pipe, "C%#o %d %s\n", mode, fileSize, fileName)
-		io.Copy(pipe, strings.NewReader(data))
-		fmt.Fprint(pipe, "\x00")
-	}()
-	if err := session.Run(fmt.Sprintf("scp -t %s", dir)); err != nil {
+	if err := session.Start(fmt.Sprintf("scp -t %s", dir)); err != nil {
 		return err
 	}
-	return nil
+	// ignore errors here. scp whould return errors if something goes wrong.
```
Comment (bprashanth) [inline test/e2e/addon_update.go]: remove this comment?

**Comment by bprashanth** [comment]
LGTM pending tests and the one nit

**Comment by bprashanth** [comment]
LGTM, feel free to apply label after you've squashed

**Comment by zmerlynn** [comment]
If you comment, too, I'll merge it ASAP.

**Comment by mikedanese** [comment]
-m --amend is not equal to -n --amend. Fixed

**Comment by zmerlynn** [comment]
Will merge on kermit.

**Comment by zmerlynn** [comment]
@mikedanese: That e2e test failure is directly related, you might want to look at it. :)

**Comment by mikedanese** [comment]
Ok, one more try. I will keep an eye and ping you if this works

**Comment by bprashanth** [comment]
The e2e failure is the notorious unable to delete script file bug https://github.com/kubernetes/kubernetes/issues/18980 though I'm just going to ask for a rerun anyway

**Comment by bprashanth** [comment]
@k8s-bot test this please

**Comment by ixdy** [comment]
@bprashanth the "unable to delete script file" isn't why the build failed; it failed because it timed out after 60 minutes (the line right before):
`Build timed out (after 60 minutes). Marking the build as failed.`

The build timed out due to #13485.

**Comment by bprashanth** [comment]
Oh, hmm, that totally draws your attention because it looks like a stack trace

**Comment by zmerlynn** [comment]
@k8s-bot e2e test this

**Comment by zmerlynn** [comment]
@ixdy: Are you sure this isn't a legit timeout? It seems pretty deterministic.

**Comment by bprashanth** [comment]
In a lot of cases some test will start and then there's a time lapse in the logs even before it runs BeforeEach:
```
07:47:34 • [SLOW TEST:12.427 seconds]
07:47:34 Services
07:47:34 /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/service.go:887
07:47:34   should serve multiport endpoints from pods [Conformance]
07:47:34   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/service.go:219
07:47:34 ------------------------------
08:21:56 SBuild timed out (after 60 minutes). Marking the build as aborted.
08:21:56 Build timed out (after 60 minutes). Marking the build as failed.
08:21:56 Terminated
```

**Comment by zmerlynn** [comment]
Yeah, that's why I was asking. It looks like how Jenkins would fail the build. (^C first (abort), then it looks like Ginkgo isn't dealing well so it fails it, IIRC?)

**Comment by zmerlynn** [comment]
In particular, even glancing through the console logs, I never see a result for `Addon update should propagate add-on file changes`, which is the test that failed previously. So I'm guessing it deadlocked.

**Comment by mikedanese** [comment]
@k8s-bot unit test this please

**Comment by zmerlynn** [comment]
LGTM

**Comment by zmerlynn** [comment]
Merging to unblock `e2e-gce`

**Pull request merged**